Example/Custom Integration/FPXExampleViewController.m

```diff
@@ -123,6 +128,27 @@ - (void)pay {
                                                }];
     }];
 }
-#pragma clang diagnostic pop
+
+- (void)bankSelectionViewController:(nonnull STPBankSelectionViewController *)bankViewController didCreatePaymentMethodParams:(STPPaymentMethodParams *)paymentMethodParams {
+    [self payWithBankAccount:paymentMethodParams.fpx.bank];
+}
+
+#pragma mark - STPAuthenticationContext
+
+- (UIViewController *)authenticationPresentingViewController {
+    return self.navigationController.topViewController;
+}
+
+- (void)authenticationContextDidPresentSafariViewController:(SFSafariViewController *)viewController {
+    // If we're launching directly into the SFSafariViewController for authentication from the bank selector,
+    // we'll want to remove the bank selector from the VC stack and pop back directly to our main controller on dismiss.
```

Comment:
Interesting problem!  Several thoughts

If we dropped it, the user would dismiss after `didCreatePaymentMethodParams`.  They could then immediately call `confirm...` ...that doesn't seem so bad, in my head - I might be missing something

---

What if the network request takes a while?  The customer would just be looking at the bank selector, right?  We should probably block them from selecting another bank.  

-----
If we keep it..

Any harm to making it `...didPresentViewController`?  Might come in handy in the future, for someone.

An alternative might be `authenticationContextWillFinish:withStatus:` or something, which would e.g. allow you to dismiss safariVC only if it was successful.  I'm thinking if a customer cancels/closes the webview, they want to go back to the previous screen (so the bank selector).

Reply:
> If we dropped it, the user would dismiss after didCreatePaymentMethodParams. They could then immediately call confirm... ...that doesn't seem so bad, in my head - I might be missing something
If they did that, they'd see the dismissal of the bank selector followed by an immediate animate-in of the SafariViewController. It doesn't look great.

> What if the network request takes a while? The customer would just be looking at the bank selector, right? We should probably block them from selecting another bank.
Yes, but they'll at least have a spinner and a disabled bank selector. It's on the app to provide more UI if they'd like.

> An alternative might be authenticationContextWillFinish:withStatus: or something, which would e.g. allow you to dismiss safariVC only if it was successful. I'm thinking if a customer cancels/closes the webview, they want to go back to the previous screen (so the bank selector).
Good idea, I'll look into that.

Reply:
> What if the network request takes a while? The customer would just be looking at the bank selector, right? We should probably block them from selecting another bank.
Yes, but they'll at least have a spinner and a disabled bank selector. It's on the app to provide more UI if they'd like.

Ooh okay, didn't realize the BankSelectorVC already disabled UI after selection.  

Everything here sounds good to me.  I bet this AuthenticationContext API will be useful in the future for some other reason.